R/lightgbm.R

```diff
@@ -247,11 +260,15 @@ train_lightgbm <- function(x,
     verbose = verbose
   )
 
-  if (!is.null(early_stop) && validation > 0) {
+  if (exists("valids")) {
     main_args$valids <- quote(valids)
+  }
+  if (!is.null(early_stop) && validation > 0) {
     main_args$early_stopping_rounds <- early_stop
   }
 
+  print("Calling lgb.train with the following args:")
+  print(main_args)
```

Comment:
I found this to be useful as I was debugging training behavior, but I'm happy to remove it if you don't think it's more widely useful.